data/source/MongoDb.php

```diff
@@ -212,53 +213,69 @@ public static function enabled($feature = null) {
 	 *        - `'database'` _string_: The name of the database to connect to. Defaults to `null`.
 	 *        - `'timeout'` _integer_: The number of milliseconds a connection attempt will wait
 	 *          before timing out and throwing an exception. Defaults to `100`.
 	 *        - `'schema'` _\Closure_: A closure or anonymous function which returns the schema
 	 *          information for a model class. See the `$_schema` property for more information.
-	 *        - `'gridPrefix'` _string_: The default prefix for MongoDB's `chunks` and `files`
-	 *          collections. Defaults to `'fs'`.
-	 *        - `'replicaSet'` _string_: See the documentation for `Mongo::__construct()`. Defaults
-	 *          to `false`.
-	 *        - `'readPreference'` _mixed_: May either be a single value such as Mongo::RP_NEAREST,
-	 *          or an array containing a read preference and a tag set such as:
-	 *          (Mongo::RP_SECONDARY_PREFERRED, ['dc' => 'east] See the documentation for
-	 *          `Mongo::setReadPreference()`. Defaults to null.
-	 *          Typically, these parameters are set in `Connections::add()`, when adding the
-	 *          adapter to the list of active connections.
 	 *
 	 *        Disables auto-connect, which is by default enabled in `Source`. Instead before
 	 *        each query execution the connection is checked and if needed (re-)established.
 	 * @return void
 	 */
 	public function __construct(array $config = []) {
-		if (class_exists($server = $this->_classes['server'], false)) {
-			$defaultHost = $server::DEFAULT_HOST . ':' . $server::DEFAULT_PORT;
-		} else {
-			$defaultHost = static::DEFAULT_HOST . ':' . static::DEFAULT_PORT;
-		}
 		$defaults = [
-			'host' => $defaultHost,
-			'login'      => null,
-			'password'   => null,
-			'database'   => null,
-			'timeout'    => 100,
-			'replicaSet' => false,
-			'schema'     => null,
-			'gridPrefix' => 'fs',
-			'w'          => 1,
-			'wTimeoutMS' => 10000,
-			'readPreference' => null,
-			'autoConnect' => false,
-			'dsn' => null
+			'host'               => static::DEFAULT_HOST . ':' . static::DEFAULT_PORT,
+			'login'              => null,
+			'password'           => null,
+			'database'           => null,
+			'timeout'            => 100,
+			'schema'             => null,
+			'uriOptions'            => [
+				'replicaSet'               => null,
+				'ssl'                      => null,
+				'connectTimeoutMS'         => null,
+				'socketTimeoutMS'          => null,
+				'maxPoolSize'              => null,
+				'minPoolSize'              => null,
+				'maxIdleTimeMS'            => null,
+				'waitQueueMultiple'        => null,
+				'waitQueueTimeoutMS'       => null,
+				'w'                        => WriteConcern::MAJORITY,
+				'wTimeoutMS'               => 10000,
+				'journal'                  => true,
+				'readConcernLevel'         => ReadConcern::LOCAL,
+				'readPreference'           => ReadPreferenceConstants::RP_PRIMARY,
```

Comment:
It's unfortunate that they don't map those constants to string values, as they do with the other ones. 

I'd prefer a small conversion method that converts the read preferences constants into strings values in a generic way (`strtolower -> camelCase`). If that utility receives string values it simply returns them unmodified. The goal is to allow users to use both string values and driver constants. 

The conversion should happen around L276, where other config mangling already takes place.

Reply:
I understand. To be honest: I don't see any value from using the constants in the provided class over [setting a string as per documentation](https://docs.mongodb.com/manual/reference/connection-string/#read-preference-options). Users will have to lookup how to set these kind of parameters, anyway. The list of available options is: **[primary, primaryPreferred, secondary, secondaryPreferred, nearest]**.

Would you be fine, if we just pass in these config-values, without "messing around™" with them?

Reply:
@davidpersson i continue with this, without messing around with these values, too much. If there are edge-cases or problems occurring, it is easy to expand the existing code to catch some easy user-errors and/or prevent common pitfalls.